### DISCUSSION TITLE
Relation Field Table Mode

### DISCUSSION THREAD

**Discussion by ChuckJonas** [keystonejs/keystone#6129] [Feature requests]
Currently the relationships support a "Card" mode to display more details of "many" records.

The current implementation is very space inefficient and I'm honestly not sure where it would ever be that useful.

<img width="570" alt="localhost_3000_coffees_ckqv5ky3m2235bx0kv1fzelyk" src="https://user-images.githubusercontent.com/5217568/125963811-4f0168b8-bc13-4dcd-9e4f-a739bfd9c674.png">

A `Data Table` seem much more useful

- Ability to define columns
- Ability to define column order
- Sorting, filtering
- Control over pagination
- Control over "unlinking" of relationship.   (maybe just if "read" mode)

Schema would look something like this:
```
      roasts: relationship({ 
        ref: 'LineItem.Order', 
        many: true,
        ui: {
          displayMode: 'table',
          tableFields: ['label', 'listPrice', 'salePrice', 'discount'],
          pageSize: 15,
        }
      }),
```

In the future, I also thing you might expand the `tableFields` prop to allow additional configuration of the column:

```
{
  field: 'category', 
  view: require.resolve('../fields/tags'),
  link: (item) => `/${item.id}/reports?category=${item.category}` 
}
```

**Comment by bladey** [comment]
Hey @ChuckJonas, thanks for the excellent detailed suggestion here, we'll take it into account!
